docs/writing-tests/testharness-tutorial.md

```diff
@@ -0,0 +1,390 @@
+# testharness.js tutorial
+
+<!--
+Note to maintainers:
+
+This tutorial is designed to be an authentic depiction of the WPT contribution
+experience. It is not intended to be comprehensive; its scope is intentionally
+limited in order to demonstrate authoring a complete test without overwhelming
+the reader with features. This should be updated as typical WPT usage patterns
+change, but please weigh extensions against the demotivating effect that a
+lengthy guide can have on new contributors.
+-->
+
```

Comment:
At this point I think we need a proper introduction before we dive in — a couple of sentences on what this tutorial teaches you; why it exists. You've sort of said that below, but not really.

Reply:
Here's the current text:

> Let's say you've discovered that WPT doesn't have any tests for how [the
> Fetch API](https://fetch.spec.whatwg.org/) sets cookies from an HTTP
> response. This tutorial will guide you through the process of writing and
> submitting a test for exactly that.

This seems to cover the concern you've mentioned, so I could use some more
detail on how to improve it. Can you expand on "You've sort of said that below,
but not really"?

Reply:
I was thinking of something like "This tutorial provides a simple introduction to writing a WebAPI test using testharness.js, verifying it, and commiting it back to WPT." The rest of the text would then follow on nicely.

Reply:
Comparing

> This tutorial provides a simple introduction to writing a WebAPI test using testharness.js, verifying it, and commiting it back to WPT.

With what's present

> This tutorial will guide you through the process of writing and submitting a test for exactly that [the fetch API].

These are similar enough to feel redundant if used together. I've tried to consolidate the two, but I've deliberately avoided some terms used by the former:

- simple (that could discourage folks that are struggling to understand this)
- WebAPI (that might be taken as jargon)
- testharness.js (that's defined by the next paragraph)

Here's what I've come up with:

> This tutorial will guide you through the process of writing a test for the web-platform, verifying it, and submitting it back to WPT.

Reply:
Great! Very well thought-out.